session_rpcserver.go

```diff
@@ -979,16 +1011,62 @@ func (s *sessionRpcServer) AddAutopilotSession(ctx context.Context,
 		caveats = append(caveats, firewall.MetaPrivacyCaveat)
 	}
 
+	// If a previous session ID has been set to link this new one to, we
+	// first check if we have the referenced session, and we make sure it
+	// has been revoked.
+	var prevSess *session.Session
+	if len(req.PrevLocalPub) != 0 {
+		prevPub, err := btcec.ParsePubKey(req.PrevLocalPub)
+		if err != nil {
+			return nil, err
+		}
+
+		prevSess, err = s.db.GetSession(prevPub)
+		if errors.Is(err, session.ErrSessionNotFound) {
+			return nil, fmt.Errorf("linked session(%x) not found",
+				req.PrevLocalPub)
+		} else if err != nil {
+			return nil, err
+		}
+
+		if prevSess.State != session.StateRevoked &&
```

Comment:
Shouldn't we check here that there's no active session for whole group, and not just that the session for the `prevPub` isn't active? Else this will allow people to set the `PrevLocalPub` to an old session in the group that has already been replaced by a new session, and this will instead fail in on the `AutoPilot` server side. I think we should have checks on both the client + server side?

Reply:
ok i've added option 2 in the commit before this one 🤓

Reply:
Awesome, thanks! Agree that this is the best option for now :)!
I guess the only case this can happen now is if a user tries to link a session to a group where all sessions are expired, but then doesn't link the latest session in the group, but a previous one. That will succeed here, but fail on the `AutoPilot` server side. Though I think that's a fair tradeoff to not add checks for that in LiT and instead just error on the `AutoPilot` server side.